Clustering/KMeans/src/main/java/org/tribuo/clustering/kmeans/KMeansTrainer.java

```diff
@@ -63,6 +67,10 @@
  * of threads used in the training step. The thread pool is local to an invocation of train,
  * so there can be multiple concurrent trainings.
  * <p>
+ * Note parallel training uses a {@link ForkJoinPool} which requires that the Tribuo codebase
+ * is given the "modifyThread" and "modifyThreadGroup" privileges when running under a
```

Comment:
I think `modifyThreadGroup` is optional for `java.lang.SecurityManager`, only needed as OpenSearch `SecureSM` has stricter check of `modifyThreadGroup`

Reply:
So I agree, but the `modifyThreadGroup` check is triggered in the codepath, it's just that `java.lang.SecurityManager` doesn't care about it. So if there are other projects which use their own subclass of `SecurityManager` might need to care about it, and explaining that level of complexity didn't seem necessary, especially as the users of Tribuo might not understand the distinction.